pkg/metric/metric.go

```diff
@@ -7,22 +7,33 @@ import (
 )
 
 var (
-	HttpDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
-		Name: "search_http_duration_seconds",
-		Help: "Latency of of HTTP requests in seconds.",
+	HttpRequestsHistogram = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "search_api_requests",
+		Help: "Histogram of HTTP requests duration (seconds).",
 	}, []string{"code"})
 )
 
 var (
 	DBConnectionFailed = promauto.NewCounterVec(prometheus.CounterOpts{
-		Name: "search_db_connection_failed_total",
+		Name: "search_api_db_connection_failed_total",
 		Help: "The total number of DB connection that has failed",
 	}, []string{"route"})
 )
 
 var (
 	DBQueryDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
-		Name: "search_dbquery_duration_seconds",
+		Name: "search_api_dbquery_duration_seconds",
 		Help: "Latency of DB requests in seconds.",
 	}, []string{"query_name"})
 )
+
+// var (
+// 	promRegistry = prometheus.NewRegistry()
+// )
+
+func GetPromRegistry() *prometheus.Registry {
+	r := prometheus.NewRegistry()
+	r.MustRegister(HttpRequestsHistogram)
+
+	return r
+}
```

Comment:
promauto will automatically register the metric when we create the histogram using that package. wondering if doing it again causes any issues?

Reply:
There's a default registry, which is what's creating the problems. The default registry has other metrics like go_* metrics which seem to add unnecessary overhead.

With the change here I create a custom registry, it only has our 3 metrics. That seems to work better but I'm not aware if it can cause other issues.

I pushed a commit to simplify this change.